### DISCUSSION TITLE
SNMP v2c How can i construct Responce message with VarBindValue::NoSuchObject

### DISCUSSION THREAD

**Discussion by JIuH4** [librasn/rasn#94] [Q&A]
SNMP v2c How can i construct Responce message with VarBindValue::NoSuchObject?
```rust
pub type Response = crate::v1::GetResponse;
```
and GetResponce accept only rasn_snmp::v1::Pdu which dont have VarBindValue::NoSuchObject

also cant parse next message from pysnmp
```rust
    let data: &[u8] = &[
        0x30, 0x3c, 0x02, 0x01, 0x01, 0x04, 0x06, 0x70, 0x75, 0x62, 0x6c, 0x69, 0x63, 0xa2, 0x2f,
        0x02, 0x04, 0x54, 0x52, 0x5d, 0x76, 0x02, 0x01, 0x00, 0x02, 0x01, 0x00, 0x30, 0x21, 0x30,
        0x0e, 0x06, 0x0a, 0x2b, 0x06, 0x01, 0x02, 0x01, 0x02, 0x02, 0x01, 0x01, 0x0c, 0x81, 0x00,
        0x30, 0x0f, 0x06, 0x0a, 0x2b, 0x06, 0x01, 0x02, 0x01, 0x02, 0x02, 0x01, 0x01, 0x0d, 0x02,
        0x01, 0x06,       
    ];
    // Decode object from BER.
    let object: Result<rasn_snmp::v2c::Message<rasn_snmp::v2::Pdus>, rasn::ber::de::Error> = rasn::ber::decode(&data);
    println!("{:?}", object);
```
it return Error 
Err(FieldError { name: "Message.data", error: "No valid `CHOICE` variant for `Pdus`" })
but must be
Ok(Message { version: 1, community: b"public", data: Response(Response(Pdu { request_id: 1414684022, error_status: 0, error_index: 0, variable_bindings: [VarBind { name: ObjectIdentifier([1, 3, 6, 1, 2, 1, 2, 2, 1, 1, 12]), value: NoSuchInstance }, VarBind { name: ObjectIdentifier([1, 3, 6, 1, 2, 1, 2, 2, 1, 1, 13]), value: Value(Simple(Integer(6))) }] })) })
it may be connected to first part of question

**Comment by XAMPPRocky** [comment]
Thank you for your question, I think the `v2` module might need to be updated to match this [RFC](https://datatracker.ietf.org/doc/html/rfc3416#section-3)
